backend/Application/Application.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Internships\Application;
+
+use Internships\FileSystem\DirectoryManager;
+use Internships\FileSystem\FileManager;
+use Internships\Services\CompanyDataBuilder;
+use Internships\Services\CsvReader;
+use Internships\Services\FacultyDataBuilder;
+use Internships\Services\UniquePathGuard;
+
+class Application
```

Comment:
Maybe we should to think about dependency injection (DI)
https://github.com/PHP-DI/PHP-DI

Reply:
I'll investigate it through separate issue.